app/models/user.js

```diff
@@ -180,18 +181,27 @@ exports.addModel = function(database) {
     return bcrypt.compareAsync(clearPassword, this.hashedPassword)
   }
 
-  User.prototype.isValidEmail = function() {
-    var valid = this.emailIsValid(this.email)
-
-    return Promise.resolve(valid)
+  User.prototype.isValidEmail = async function() {
+    return User.emailIsValid(this.email)
   }
 
-  User.prototype.emailIsValid = function(email) {
+  User.emailIsValid = async function(email) {
```

Comment:
Let's extract to Validator module, which does not exist yet. Could be in a separate PR.